draft-ietf-quic-version-negotiation.md

```diff
@@ -67,105 +67,171 @@ Discussion of this work is encouraged to happen on the QUIC IETF mailing list
+
+
+## Incompatible Version Negotiation {#incompat-vn}
+
+The server starts incompatible version negotiation by sending a VN packet,
+listing all the versions that it does support.
+
+Upon receiving the VN packet, the client will search for a version it supports
+in the list provided by the server. If it doesn't find one, it aborts the
+connection attempt. Otherwise, it selects a mutually supported version and
+sends a new first flight with that version - we refer to this version as the
+"negotiated version".
+
+When the client sends its new first flight, it SHOULD use the same 5-tuple
+(source and destination IP addresses and UDP port numbers) and the same
+Destination Connection ID as its original first flight. This will increase the
+likelihood of both first flights reaching the same server.
+
+The new first flight will allow the endpoints to establish a connection using
+the negotiated version. The handshake of the negotiated version will exchange
+handshake version information (see {{hs-vers-info}}) required to ensure that VN
+was genuine, i.e. that no attacker injected packets in order to influence the
+VN process, see {{downgrade}}.
+
+
+## Compatible Version Negotiation {#compat-vn}
+
+When the server can parse the client's first flight using the original version,
+it can extract the client's handshake version information (see
+{{hs-vers-info}}). This contains the list of versions that the client thinks
+its original version is compatible with.
+
+If the server supports one of the client's compatible versions, and the server
+also believes that the original version is compatible with this version, then
+the server attempts to convert the client's first flight to that version. If
+that succeeds, the server replies to the client as if it had received the
+converted first flight. The version used by the server in its reply is refered
+to as the "negotiated version". The server MUST NOT reply with a version that
+is not present in the client's compatible versions, unless it is the original
+version.
+
+If any of these operations fail (except parsing and validation failures which
+are discussed in {{downgrade}}), the server will use the original version if it
+supports it, and if it doesn't then the server will perform incompatible
+version negotiation instead, see {{incompat-vn}}.
+
+For the duration of the compatible version negotiation process, clients MUST
+use the same 5-tuple (source and destination IP addresses and UDP port
+numbers). During that time, clients MUST also use the same Destination
+Connection ID, except if the server explicitly instructs the client to use a
+different Destination Connection ID. This allows load balancers to ensure that
+packets for a given connection are routed to the same server.
+
+
+# Handshake Version Information {#hs-vers-info}
+
+During the handshake, endpoints will exchange handshake version information,
+which is an opaque blob of data that is defined below. In QUIC version 1, the
```

Comment:
How is this opaque?

Reply:
In that versions themselves don't need to worry about what's inside?